newIDE/app/src/Utils/GDevelopServices/Asset.js

```diff
@@ -328,3 +332,30 @@ export const listReceivedAssetPacks = async (
   });
   return response.data;
 };
+
+export const isPublicAssetResourceUrl = (url: string) =>
+  url.startsWith(GDevelopPublicAssetResourcesStorageBaseUrl) ||
+  url.startsWith(GDevelopPublicAssetResourcesStorageStagingBaseUrl);
+const escapeStringForRegExp = string =>
+  string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+const resourceFilenameRegex = new RegExp(
+  `(${escapeStringForRegExp(
+    GDevelopPublicAssetResourcesStorageBaseUrl
+  )}|${escapeStringForRegExp(
+    GDevelopPublicAssetResourcesStorageStagingBaseUrl
+  )})\\/public-resources\\/(.*)\\/([a-z0-9]{64})_(.*)(\\..*)`
+);
+export const extractFilenameAndExtensionFromPublicAssetResourceUrl = (
```

Comment:
Something that I wanted to do in my still not merged PR about cloud projects on desktop:
don't return "filename and extension", just return the filename with the extension (`myfile.png`).

The caller can extract the extension if they wish (but it's not always the case, at one point the code is concatenating them just after getting them!)

Reply:
true, I'll change that.